hadoop-plugin/src/main/groovy/com/linkedin/gradle/test/TestPlugin.groovy

```diff
@@ -0,0 +1,279 @@
+   *
+   * @param project The Gradle project
+   * @return The AzkabanDslCompiler
+   */
+  AzkabanDslCompiler makeCompiler(Project project) {
+    return new AzkabanDslCompiler(project);
+  }
+
+  /**
+   * This task will take all the job files along with the hadoopZip extension and create a separate zip for testing of the project. Technically we
+   * want to retain the original zip but for each test a new zip has to be generated.
+   * @param project The hadoop project
+   * @return Return the created task
+   */
+  Task createZipForTestingTask(Project project) {
+    return project.tasks.create("buildZipForTesting") {
+      dependsOn project.tasks["buildHadoopZips"]
+      dependsOn project.tasks["buildFlowsForTesting"]
+      project.tasks["buildFlowsForTesting"].shouldRunAfter project.tasks["buildAzkabanFlows"]
+      project.tasks["buildHadoopZips"].shouldRunAfter project.tasks["buildFlowsForTesting"]
+    }
+  }
+
+  /**
+   * Creates the testDeploy task. This task will automatically build and upload the zip to the test machines
+   * @param project The Gradle project
+   * @return The created task
+   */
+  Task createTestDeployTask(Project project) {
+    return project.task("testDeploy", type: AzkabanUploadTask) { task ->
+      description = "Deploys the test provided by -Ptestname=name"
+      group = "Hadoop Plugin";
+      dependsOn project.tasks["buildZipForTesting"]
+
+      doFirst {
+
+        if (!project.hasProperty("testname")) {
+          logger.error("You must specify the name of the test using -Ptestname=name");
+          throw new RuntimeException("Test not found");
+        }
+
+        if (project.hasProperty("skipInteractive")) {
+          logger.lifecycle("Skipping interactive mode");
+          interactive = false;
+        }
+
+        azkProject = readAzkabanProject(project);
+
+        String zipTaskName = azkProject.azkabanZipTask;
+        if (!zipTaskName) {
+          throw new GradleException("\nPlease set the property 'azkabanZipTask' in the .azkabanPlugin.json file");
+        }
+
+        def zipTaskCont = project.getProject().tasks[zipTaskName];
+        if (zipTaskCont == null) {
+          throw new GradleException(
+              "\nThe task ${zipTaskName} doesn't exist. Please specify a Gradle Zip task after configuring it in your build.gradle file.");
+        }
+
+        if (!zipTaskCont instanceof Zip) {
+          throw new GradleException(
+              "\nThe task ${zipTaskName} is not a Zip task. Please specify a Gradle Zip task after configuring it in your build.gradle file.");
+        }
+
+        archivePath = zipTaskCont.archivePath;
+      }
+    }
+  }
+
+  /**
+   * Helper method to determine the location of the plugin json file. This helper method will make
+   * it easy for subclasses to get (or customize) the file location.
+   *
+   * @param project The Gradle project
+   * @return The path to the plugin json file
+   */
+  String getPluginJsonPath(Project project) {
```

Comment:
The readAzkabanProject is different from the readAzkabanProject in AzkabanHelper. For the other two methods, they are not really utility methods, they are infact part of the class definition. In the later PR, I'll change the default project and the json file for tests.